server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolParser.g4

```diff
@@ -25,130 +25,137 @@ endProgramStatement
 
 // compiler options
 compilerOptions
-   : (PROCESS | CBL) (COMMACHAR? compilerOption | compilerXOpts)+
-   ;
-
-compilerXOpts
-   : XOPTS LPARENCHAR compilerOption (COMMACHAR? compilerOption)* RPARENCHAR
+   : (PROCESS | CBL) (COMMACHAR? compilerOption)+
    ;
 
+// https://www.ibm.com/docs/en/cobol-zos/6.3?topic=program-compiler-options
 compilerOption
-   : ADATA | ADV | APOST | AWO
+   : ADATA | NOADATA
+   | ADV | NOADV
+   | AFP LPARENCHAR (NOVOLATILE | VOLATILE) RPARENCHAR
+   | APOST | QUOTE | Q_CHAR
+   | ARCH LPARENCHAR integerLiteral RPARENCHAR
    | (ARITH | AR) LPARENCHAR (EXTEND | E_CHAR | COMPAT | C_CHAR) RPARENCHAR
-   | BLOCK0
-   | (BUFSIZE | BUF) LPARENCHAR literal RPARENCHAR
-   | CBLCARD
-   | CICS (LPARENCHAR literal RPARENCHAR)?
-   | COBOL2 | COBOL3
-   | (CODEPAGE | CP) LPARENCHAR literal RPARENCHAR
-   | (COMPILE | C_CHAR)
-   | CPP | CPSM
-   | (CURRENCY | CURR) LPARENCHAR literal RPARENCHAR
-   | DATA LPARENCHAR literal RPARENCHAR
-   | (DATEPROC | DP) (LPARENCHAR (FLAG | NOFLAG)? COMMACHAR? (TRIG | NOTRIG)? RPARENCHAR)?
-   | DBCS
-   | (DECK | D_CHAR)
-   | DEBUG
-   | (DIAGTRUNC | DTR)
-   | DLL
-   | (DUMP | DU)
-   | (DYNAM | DYN)
-   | EDF | EPILOG
-   | EXIT
-   | (EXPORTALL | EXP)
-   | (FASTSRT | FSRT)
-   | FEPI
+   | AWO | NOAWO
+   | BLOCK0 | NOBLOCK0
+   | (BUFSIZE | BUF) LPARENCHAR (integerLiteral | INTEGERLITERAL_WITH_K) RPARENCHAR
+   | CICS (LPARENCHAR literal RPARENCHAR)? | NOCICS
+   | (CODEPAGE | CP) LPARENCHAR integerLiteral RPARENCHAR
+   | COMPILE | C_CHAR | (NOCOMPILE | NOC) (LPARENCHAR (S_CHAR | E_CHAR | W_CHAR) RPARENCHAR)?
+   | (COPYLOC | CPLC) LPARENCHAR ((SYSLIB | dataName) COMMACHAR)? (DSN | PATH) LPARENCHAR (literal | dataSetName) RPARENCHAR RPARENCHAR
+   | NOCOPYLOC | NOCPLC
+   | (COPYRIGHT | CPYR) LPARENCHAR literal RPARENCHAR | NOCOPYRIGHT | NOCPYR
+   | (CURRENCY | CURR) LPARENCHAR . RPARENCHAR | NOCURRENCY | NOCURR
```

Comment:
Does it allow anything inside the parenthesis?

Reply:
Yes, there can be any character like `€`, `$` or `₽`. And I have no idea how to do it the other way.

Reply:
Did you try this? https://stackoverflow.com/questions/14169820/regular-expression-to-match-all-currency-symbols/14169836

Reply:
Yes, you are right.